plugins/redfish/fu-redfish-network.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-redfish-network.h"
+
+typedef struct {
+	gchar		*address;
+	gboolean	 device_found;
+	const gchar	*mac_addr;
+	guint16		 vid;
+	guint16		 pid;
+} FuRedfishNetworkMatchHelper;
+
+static gboolean
+fu_redfish_network_ip_match_ip4_config (FuRedfishNetworkMatchHelper *helper,
+					const gchar *object_path,
+					GError **error)
+{
+	g_autoptr(GDBusProxy) proxy = NULL;
+	g_autoptr(GVariant) addr_data = NULL;
+
+	proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+					       G_DBUS_PROXY_FLAGS_DO_NOT_CONNECT_SIGNALS,
+					       NULL,
+					       "org.freedesktop.NetworkManager",
+					       object_path,
+					       "org.freedesktop.NetworkManager.IP4Config",
```

Comment:
network manager on a server?  Is this super common?

Reply:
On RHEL, yes, it's a hard dep. On others not sure, and I'd be totally okay if someone wanted to do a netlink-style implementation if they wanted; I just don't want that person to be me :)